usage_reporter.py

```diff
@@ -0,0 +1,170 @@
+# Copyright 2023 Peter Bentley.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import csv
+import os
+import re
+import sys
+
+import requests
+import urllib.parse
+from datetime import datetime
+from typing import Optional
+
+# Usage: python usage_reporter.py [update]
+#
+# If no update argument is given, downloads stats for the previous calendar month and
+# displays them.  NB relies on the fact that if only a single month is requested, the data
```

Comment:
Maybe a dumb question but what is "NB" in this context?

Reply:
I think it's https://en.wikipedia.org/wiki/Nota_bene

Reply:
Will reword!

Reply:
Reworded.